### PR TITLE
Ocp wo json

### PR DESCRIPTION
**Pull request by ikegami-t**

### CODE REVIEW

**Comment by igaw** [comment]
I was wondering if we should move the json code into a separate files and only include them when the json library is available, like we have with `nvme-print-stdout.c` and `nvme-print-json.c`.

**Comment by ikegami-t** [comment]
Okay will try to do it. Thank you.

File: plugins/ocp/meson.build [hunk 1]
```diff
@@ -6,5 +6,9 @@ sources += [
   'plugins/ocp/ocp-fw-activation-history.c',
   'plugins/ocp/ocp-telemetry-decode.c',
   'plugins/ocp/ocp-hardware-component-log.c',
+  'plugins/ocp/ocp-print.c',
+  'plugins/ocp/ocp-print-stdout.c',
+  'plugins/ocp/ocp-print-json.c',
```
Comment (igaw) [inline plugins/ocp/meson.build:11]: The json file should only be added when the json library is available. And then you could update the `plugins/meson.build` file to always build the ocp plugin.

The rest looks good!

Comment (ikegami-t) [reply plugins/ocp/meson.build]: Update the meson.build files by the additional commit. Thank you.
  Note: Still some exisitng ocp commands files needed to split json code.

Comment (igaw) [reply plugins/ocp/meson.build]: So do you want me to wait with merging this and wait until you finished the json code splitting?

Comment (ikegami-t) [reply plugins/ocp/meson.build]: No need to wait for me. Still working on this but I am okay to do it by additional PRs. Thank you.

**Comment by ikegami-t** [comment]
All OCP command json codes execpt the ocp-telemetry-decode functions are split into the ocp-print-json.c.
Note: The ocp-telemetry-decode functions used by the internal command are mixed stdout and json outputs implementation so not moved into the ocp-print-json.c but using the CONFIG_JSONC to disable intead.

**Comment by ikegami-t** [comment]
Just rebased and fixed the conflicts.

**Comment by igaw** [comment]
I've noticed that the stdout plugin does

```c
static struct ocp_print_ops stdout_print_ops = {
	.hwcomp_log = stdout_hwcomp_log,
	.fw_act_history = stdout_fw_activation_history,
	.smart_extended_log = stdout_smart_extended_log,
	.telemetry_log = stdout_telemetry_log,
	.c3_log = (void *)stdout_c3_log,
	.c5_log = (void *)stdout_c5_log,
	.c1_log = stdout_c1_log,
	.c4_log = stdout_c4_log,
	.c9_log = (void *)stdout_c9_log,
	.c7_log = (void *)stdout_c7_log,
};
```

I think the cast should be removed by updating the function or updating the callback function signature. Either way, cast should go away eventually. Though no reason not go apply it now. Thanks for the great work! Really appreciated!

**Comment by ikegami-t** [comment]
Thank you so much for your reviewing and comments. Just fixed for the comment by the PR #2541. Thank you.

**Pull request merged**